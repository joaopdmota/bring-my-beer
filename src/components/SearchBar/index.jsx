// @flow
import React, { useState, useEffect } from 'react'
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import styled from 'styled-components'
import Input from '../Input'
import IcoSearch from '../../assets/search.svg'
import IcoPin from '../../assets/pin.svg'
import Grid from '../Grid'
import { getLocation } from '../../utils/location'
import { setPosition, getProducts } from '../../actions'

const SearchBar = styled(Grid)`
  width: 100%;
  input {
    flex-grow: 2;
    margin-bottom: 0px;
  }
  svg {
    cursor: pointer;
    margin-left: 8px;
  }
`

type SearchBarProps = {
  doSearch: string => void,
  hasPin: boolean,
  placeholder: string,
}

export default ({
  doSearch,
  hasPin = true,
  placeholder = 'digite seu endereÃ§o aqui',
}: SearchBarProps) => {
  const [currentLocation, setCurrentLocation] = useState([])
  const [searchText, setSearchText] = useState('')
  const history = useHistory()
  const dispatch = useDispatch()

  const handleChange = e => {
    setSearchText(e.target.value)
  }

  const handleKey = e => {
    if (e.keyCode === 13) {
      doSearch(searchText)
    }
  }

  const handlePin = e => {
    getLocation(coordinates => {
      setCurrentLocation([
        coordinates.coords.latitude,
        coordinates.coords.longitude,
      ])
    })
  }

  useEffect(() => {
    if (currentLocation.length) {
      dispatch(setPosition(currentLocation))
      history.push('/results')
    }
  }, [currentLocation])

  return (
    <SearchBar>
      <Input
        placeholder={placeholder}
        onKeyDown={handleKey}
        value={searchText}
        onChange={handleChange}
      />
      <IcoSearch height={35} onClick={() => doSearch(searchText)} />
      {hasPin && <IcoPin height={35} onClick={handlePin} />}
    </SearchBar>
  )
}
