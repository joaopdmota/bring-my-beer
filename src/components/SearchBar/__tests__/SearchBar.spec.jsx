import React from 'react'
import { shallow, mount } from 'enzyme'
import SearchBar from '../index'

const doSearch = jest.fn()
const onChange = jest.fn()
const doAfterSearch = jest.fn()

describe('SearchBar Component', () => {
  it('Render default SerchBar', () => {
    const searchbar = shallow(<SearchBar onChange={onChange} value="" />)
    expect(searchbar).toMatchSnapshot()
  })

  it('Render SearchBar with no pin', () => {
    const searchbar = shallow(
      <SearchBar hasPin={false} onChange={onChange} value="" />,
    )
    expect(searchbar).toMatchSnapshot()
  })
})
