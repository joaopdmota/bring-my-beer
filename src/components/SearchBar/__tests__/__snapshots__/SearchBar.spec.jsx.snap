// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component Render SearchBar with no pin 1`] = `
<SearchBar>
  <_default
    error={false}
    message=""
    onChange={[MockFunction]}
    onKeyDown={[Function]}
    placeholder="digite seu endereço aqui"
    value=""
  />
  <search.svg
    height={35}
    onClick={[Function]}
  />
</SearchBar>
`;

exports[`SearchBar Component Render default SerchBar 1`] = `
<SearchBar>
  <_default
    error={false}
    message=""
    onChange={[MockFunction]}
    onKeyDown={[Function]}
    placeholder="digite seu endereço aqui"
    value=""
  />
  <search.svg
    height={35}
    onClick={[Function]}
  />
  <pin.svg
    height={35}
    onClick={[Function]}
  />
</SearchBar>
`;
